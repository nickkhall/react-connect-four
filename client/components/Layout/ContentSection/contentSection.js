import React from 'react';

// MUI Components
import Paper from '@material-ui/core/Paper';

export default ({ children, classes }) => (
  <div className={classes.verticalScroll}>
    <Paper className={classes.paperDefault}>{children}</Paper>
  </div>
);
