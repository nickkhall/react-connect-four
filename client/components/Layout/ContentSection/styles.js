export default theme => ({
  paperDefault: {
    margin: theme.spacing.unit * 2,
    padding: theme.spacing.unit * 2,
    textAlign: 'left',
    color: theme.palette.text.secondary,

    [theme.breakpoints.up('sm')]: {
      minWidth: '400px'
    }
  },
  verticalScroll: {
    overflowY: 'auto'
  }
});
