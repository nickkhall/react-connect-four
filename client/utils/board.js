export const getNextNum = (val, amount) => {
  const splitVal = val.split('');
  const nextLetter = String.fromCharCode((parseInt(splitVal[0].charCodeAt(0), 10) + amount));
  const nextNum = parseInt(splitVal[1], 10) - amount;

  return `${nextLetter}${nextNum}`;
};
