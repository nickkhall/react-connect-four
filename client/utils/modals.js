/**
 * Gets the active modal (if available) from the modal stack. In some situations
 * we have multiple active modals (such as when we open an image select or
 * vendor modal on top of an edit modal).
 * @param {!Array.<!Object>} stack The modal stack.
 * @return {{name: string, props: Object}} The current modal object.
 */
export const getActiveModal = stack => (stack && stack.length ? stack[stack.length - 1] : {});
