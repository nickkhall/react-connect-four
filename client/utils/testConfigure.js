/* eslint-disable import/no-extraneous-dependencies, no-console */
// React 16.x requires an Enzyme adapter.
// See: http://airbnb.io/enzyme/docs/installation/index.html#installation
// And: http://facebook.github.io/jest/docs/en/configuration.html#setuptestframeworkscriptfile-string
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

if (!global.TEST_CFG) {
  global.TEST_CFG = true;

  configure({ adapter: new Adapter() });

  const mockNoop = () => new Promise(() => {});

  jest.mock('axios', () => ({
    default: mockNoop,
    get: mockNoop,
    post: mockNoop,
    put: mockNoop,
    delete: mockNoop,
    patch: mockNoop
  }));

  jest.mock('store', () => ({
    dispatch: jest.fn()
  }));

  process.on('unhandledRejection', (e) => {
    console.error(e);
  });
}
