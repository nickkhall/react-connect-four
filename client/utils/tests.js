import React from 'react';

/* eslint-disable import/no-extraneous-dependencies */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

/* eslint-disable no-console */
export const mockNotificationDud = { type: 'MOCK_NOTIF_DUD' };

export const testingFunc = test => `should contain callable ${test} function`;

export const handleTestError = (test, err, done) => {
  console.warn('Error with', test);
  console.error(new Error(err));
  done();
};

/**
 * Automatically tests that the component matches the Jest snapshot.
 * NOTE: If you are receiving a warning like the following in your tests:
 *
 *   Warning: React.createElement: type is invalid -- expected a string...
 *
 * Then you most likely need to pass the appropriate props.
 *
 * @param {!React.Component} Component The React component to wrap and test.
 * @param {!Object} props The (optional) props to use for the basic Jest test.
 */
export const testCommonComponentAttrs = (Component, props) => {
  describe('when rendering', () => {
    const wrapper = shallow(<Component {...props} />);

    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
    });
  });
};

export const testDispatchToProps = (Component, mapDispatchToProps) => {
  let mockDispatch;
  let mockActions;

  beforeEach(() => {
    mockDispatch = jest.fn();
    mockActions = mapDispatchToProps(mockDispatch);
  });

  describe('when mapping dispatch to props', () => {
    const actions = Object.keys(mapDispatchToProps());
    actions.forEach((action) => {
      it(`should map the ${action} action`, () => {
        expect(mockActions[action]).toBeDefined();
        expect(typeof mockActions[action]).toEqual('function');
        mockActions[action]();
        expect(mockDispatch).toHaveBeenCalled();
      });
    });
  });
};

export const getComponentWrapper = (Component, props) => shallow(<Component {...props} />);


export const testReducerDefaults = (reducer, initialState) => {
  describe('Reducer Defaults', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(initialState);
    });
  });
};
