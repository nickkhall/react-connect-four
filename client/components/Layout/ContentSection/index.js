// MUI Components
import { withStyles } from '@material-ui/core/styles';

// Components
import ContentSection from './contentSection';

// Local styles
import styles from './styles';

export default withStyles(styles)(ContentSection);
