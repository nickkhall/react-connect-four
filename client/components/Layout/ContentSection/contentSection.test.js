// Test utils
import { testCommonComponentAttrs } from 'utils/tests';

import ContentSection from './contentSection';

const defaultProps = { classes: {} };

describe('<ContentSection /> component', () => {
  testCommonComponentAttrs(ContentSection, defaultProps);
});
