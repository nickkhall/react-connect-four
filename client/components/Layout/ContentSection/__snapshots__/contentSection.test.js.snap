// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentSection /> component when rendering should render a component matching the snapshot 1`] = `
<div>
  <WithStyles(Paper) />
</div>
`;
